src/main/java/org/vanilladb/core/storage/metadata/index/IndexMgr.java

```diff
@@ -145,4 +146,65 @@ public void createIndex(String idxName, String tblName, String fldName,
 		iiMap.put(tblName, result);
 		return result;
 	}
+
+	/**
+	 * Returns the requested index.
+	 * 
+	 * @param idxName
+	 *            the name of the index
+	 * @param tx
+	 *            the calling transaction
+	 * @return an Index object
+	 */
+	public Index getIndex(String idxName, Transaction tx) {
```

Comment:
Please name this method as `getIndexInfoByName` and make it return an `IndexInfo`. Since opening an index requires additional I/O operations, we should leave this decision to the clients calling this method.